> This is a real bug fix, and people observing the weird behavior with old versions of OCaml would appreciate to see an explicit reference to the fix in the Changelog.

OK, I'll add a Changes entry.

> And it would indeed be interesting to understand why there was no compiler warning!

`cl` is not careful enough by default. I compiled the following program
```c
/* foo.c */
#include <wchar.h>
#include <stdio.h>

int main(void)
{
  wchar_t foo[2];
  char c = foo[0];
  printf("%c\n", c); fflush(stdout);
  return 0;
}
```
with `cl /Wn foo.c` with `n = 0, 1, 2` (no warning). Only with `n >= 3` (`n = 4` is the maximum) did I get
```
foo.c(7): warning C4244: 'initializing': conversion from 'wchar_t' to 'char', possible loss of data
```
I will try to see if it is possible to compile the runtime system with `/W3` later.

--- a/Changes
+++ b/Changes
@@ -273,6 +273,10 @@ Working version
   to work on ocamltest-based tests.
   (Runhang Li and Sébastien Hinderer, review by Gabriel Scherer)
 
+- GPR#1622: fix bug in the expansion of command-line arguments under Windows
+  which could result in some elements of Sys.argv being truncated in some cases.
+  (Nicolás Ojeda Bär, review by Sébastien Hinderer)
+
 4.06 maintenance branch
 -----------------------
 

